server/server.go

```diff
@@ -171,6 +173,19 @@ func (s *Server) genericOAuth(logger chronograf.Logger, auth oauth2.Authenticato
 	return &gen, genMux, s.UseGenericOAuth2
 }
 
+func (s *Server) genericRedirectURL() string {
+	if s.PublicURL == "" {
+		return ""
+	}
+
+	genericName := "generic"
+	if s.GenericName != "" {
+		genericName = s.GenericName
+	}
+
+	return path.Join(s.PublicURL, s.Basepath, "oauth", genericName, "callback")
```

Comment:
I hate to ask you to change this again, but there's actually an issue with joining URLs like this.

`path.Join` will take any duplicate slashes found within path segments and de-duplicate them, so if the `s.PublicURL` contains the URL scheme, it'll change `http://` to `http:/` and make the URL invalid. You can see this in [this Go playground](https://play.golang.org/p/gmIBzMCyvX)

The safest way I've found to build URLs up is to parse the `s.PublicURL` using `url.Parse` (so that `http` vs. `https` is handled for you), and then set the `Path` attribute of that with `path.Join`. Then you can call `String()` on the whole thing to get the full URL.